cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/AnalyzerState.java

```diff
@@ -37,16 +39,28 @@ public boolean proposalReady() {
    * Return an object that can be further used
    * to encode into JSON
    */
-  public Map<String, Object> getJsonStructure() {
+  public Map<String, Object> getJsonStructure(boolean verbose) {
     Map<String, Object> analyzerState = new HashMap<>();
     Set<String> readyGoalNames = new HashSet<>();
+    List<Object> goalReadinessList = new ArrayList<>();
     for (Map.Entry<Goal, Boolean> entry : _readyGoals.entrySet()) {
+      Goal goal = entry.getKey();
+      Map<String, Object> goalReadinessRecord = new HashMap<>();
+      goalReadinessRecord.put("name", goal.getClass().getSimpleName());
+      goalReadinessRecord.put("modelCompleteRequirement", goal.clusterModelCompletenessRequirements().getJsonStructure());
       if (entry.getValue()) {
         readyGoalNames.add(entry.getKey().name());
+        goalReadinessRecord.put("Status", "Ready");
+      } else {
+        goalReadinessRecord.put("Status", "NotReady");
       }
+      goalReadinessList.add(goalReadinessRecord);
     }
     analyzerState.put("isProposalReady", _isProposalReady);
     analyzerState.put("readyGoals", readyGoalNames);
+    if (verbose) {
```

Comment:
Can we move the complete building of `goalReadinessList` under the verbose condition here? My thought here is that the verbose option will be less frequent and we are generating `goalRedinessList` for every non-verbose option. IMO addition for-loop for verbose should be okay.